enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py

```diff
@@ -94,53 +99,48 @@ def _manager_on_boot(self):
-        - process_wo_id is id of the work-order that is to be processed
+            @param process_wo_id - Id of the work-order that is to be processed
         """
-        logger.info("Processing work order")
-        try:
-            # Get all workorders requests from KV storage lookup and process
-            list_of_workorders = self._kv_helper.lookup("wo-scheduled")
-            if not list_of_workorders:
-                logger.info("Received empty list of work orders from " +
-                            "wo-scheduled table")
-                return
-        except Exception as e:
-            logger.error("Problem while getting ids from wo-scheduled table")
-            return
-
-        for wo_id in list_of_workorders:
-            if wo_id == process_wo_id:
-                resp = self._process_work_order_by_id(wo_id)
-                return resp
+        logger.info("Processing work orders found in wo-worker-pending table.")
+
+        wo_id = self._kv_helper.csv_pop("wo-worker-pending", self._worker_id)
+        while wo_id is not None:
+            # @TODO : csv_pop needs to be removed for cases of multiple WPE in
+            # a pool. The entry popped needs to be reverted/put back in case of
+            # a mismatch else it will be lost. As of now, even in pool mode a
+            # single WPE per pool is supported for sync mode using ZMQ.
+            if process_wo_id == wo_id:
+                self._kv_helper.set("wo-worker-map",
+                                    self._enclave_id_hex, wo_id)
+                return self._process_work_order_by_id(wo_id)
             else:
                 return None
-        # end of for loop
+            wo_id = self._kv_helper.csv_pop(
+                "wo-worker-pending", self._worker_id)
+        # end of loop
+
+        logger.info("No more worker orders in wo-worker-pending table.")
 
 # -------------------------------------------------------------------------
 
     def _process_work_orders(self):
         """
         Executes Run time flow of enclave manager
         """
-        logger.info("Processing work orders")
-        try:
-            # Get all workorders requests from KV storage lookup and process
-            list_of_workorders = self._kv_helper.lookup("wo-scheduled")
-            if not list_of_workorders:
-                logger.info("Received empty list of work orders from " +
-                            "wo-scheduled table")
-                return
-        except Exception as e:
-            logger.error("Problem while getting keys from wo-scheduled table")
-            return
+        logger.info("Processing work orders found in wo-worker-pending table.")
 
-        for wo_id in list_of_workorders:
+        wo_id = self._kv_helper.csv_pop("wo-worker-pending", self._worker_id)
+        while wo_id is not None:
+            self._kv_helper.set("wo-worker-map",
+                                self._enclave_id_hex, wo_id)
```

Comment:
Can we use self._worker_id instead of self._enclave_id_hex to be consistent with singleton

Reply:
You could remove `wo-worker-map` from this PR as it's not getting used

Reply:
> Can we use self._worker_id instead of self._enclave_id_hex to be consistent with singleton

_worker_id is explicitly needed by WPE as well.